astropy/units/format/base.py

```diff
@@ -312,3 +303,7 @@ def _decompose_to_known_units(
         raise TypeError(
             f"unit argument must be a 'NamedUnit' or 'CompositeUnit', not {type(unit)}"
         )
+
+    @classmethod
+    def _validate_named_unit(cls, unit: NamedUnit) -> None:
```

Comment:
Naming is confusing here! `_validate_unit` validates a string argument, while `_validate_named_unit` validates a `NamedUnit`.

Perhaps, the former should be `_get_unit_by_name` or `_unit_from_name` and the latter can become `_validate_unit`?

Though these are private methods, a short docstring might also help...